module/module.go

```diff
@@ -0,0 +1,422 @@
+package module
```

Comment:
given this is basically the SDK part of module, would it be better to call this the module client/child given we have the word parent? I want to make it obvious this file is for module writers

Reply:
I think that looks pretty clear once we have examples/tutorials. A module author will only ever need to see/use this package (not modmanager and others) so "import blah/blah/module" looks pretty intuitive to me, if I've set out to write a module. Least that's my gut feeling on the name.

Reply:
alright